makeReducedConfig.py

```diff
@@ -160,37 +160,61 @@ def addTC(self, tc, layers, ref_p, noneg):
         l1phis = ["A", "B", "C", "D", "E", "F", "G", "H"]
         lxphis = ["A", "B", "C", "D"]
 
-        if tc.upper() not in tcs:
-            print("Bad " + module_type + " index, not adding.")
-            return
-
         # Store whether or not we're including negative eta inputs
         self.noneg = noneg
 
         # Store which layer pair we're seeding from to use when deciding what to keep
         self.seed_layers.append(layers[:2])
         self.seed_layers.append(layers[2:])
 
-        tc_i = tcs.index(tc.upper())
-        phi1_i = int(tc_i*1.*len(l1phis)/len(tcs))
-        phix_i = int(tc_i*1.*len(lxphis)/len(tcs))
-
-        print("Adding regions to project: %s_%s%s, L1PHI%s, LxPHI%s"%(module_type, layers, tcs[tc_i], l1phis[phi1_i], lxphis[phix_i]))
-        self.tcs.append("%s_%s%s"%(module_type, layers, tcs[tc_i]))
-        self.l1phis.append(l1phis[phi1_i])
-        self.lxphis.append(lxphis[phix_i])
 
-        # Add nodes and connections to the project
-        # Starting with e.g. TC_L1L2F and moving up and down the chain
-        # For each node it adds, will check for all inputs and outputs of that node
-
-        print("Starting with node %s_%s%s"%(module_type,layers,tcs[tc_i]))
-        n = node("%s_%s%s"%(module_type,layers,tcs[tc_i]))
-        self.addNode(n)
-        print("Finding inputs...")
-        self.findInputConnections(n, ref_p)
-        print("Finding outputs...")
-        self.findOutputConnections(n, ref_p)
+        if tc == "All":
+            for tc in tcs:
+                tc_i = tcs.index(tc.upper())
+                phi1_i = int(tc_i*1.*len(l1phis)/len(tcs))
+                phix_i = int(tc_i*1.*len(lxphis)/len(tcs))
+
+                print("Adding regions to project: %s_%s%s, L1PHI%s, LxPHI%s"%(module_type, layers, tcs[tc_i], l1phis[phi1_i], lxphis[phix_i]))
+                self.tcs.append("%s_%s%s"%(module_type, layers, tcs[tc_i]))
+                self.l1phis.append(l1phis[phi1_i])
+                self.lxphis.append(lxphis[phix_i])
+
+                # Add nodes and connections to the project
+                # Starting with e.g. TC_L1L2F and moving up and down the chain
+                # For each node it adds, will check for all inputs and outputs of that node
+
+                print("Starting with node %s_%s%s"%(module_type,layers,tcs[tc_i]))
+                n = node("%s_%s%s"%(module_type,layers,tcs[tc_i]))
+                self.addNode(n)
+                print("Finding inputs...")
+                self.findInputConnections(n, ref_p)
+                print("Finding outputs...")
+                self.findOutputConnections(n, ref_p)
```

Comment:
@aryd Since this code is identical to most of the code in the `else` block, could you break it off into a separate method to reduce code duplication?

Reply:
I rewrote the code to avoid this duplication.